lib/sinatra/base.rb

```diff
@@ -862,7 +876,10 @@ def render(engine, data, options = {}, locals = {}, &block)
         catch(:layout_missing) { return render(layout_engine, layout, options, locals) { output } }
       end
 
-      output.extend(ContentTyped).content_type = content_type if content_type
+      if content_type
+        output = +output
```

Comment:
Was this a problem before, that this string was frozen?

Maybe we should commit this change separately? With a test for it?

Reply:
This is due to google-protobuf gem decodes the output as frozen string. There is a performance concern on eagerly unfreezing when not necessary. Therefore, sass-embedded returns the css output as frozen string as is, for which dependencies need to unfrozen the string as needed.

Reply:
Can we only unfreeze when sass-embedded is used? Can we at least add a comment to the code with the above information?

Reply:
I added a comment. 

As for the `+string` method:
> Returns self if self is not frozen.
> Otherwise. returns self.dup, which is not frozen.

I did a microbenchmark on different ways to write this, and `output = +output` is the most performant one:

```
       user     system      total        real
s.freeze; s = s.dup if s.frozen?;	  1.553977   0.006994   1.560971 (  1.561001)
s.freeze; s = +s if s.frozen?;		  0.919195   0.002047   0.921242 (  0.921248)
s.freeze; s = +s;			  0.763196   0.001567   0.764763 (  0.764776)
```